### PR TITLE
Consolidate workflow groupings

### PR DESCRIPTION
**Pull request by tstirrat**
Consolidate workflow groups into:

- Build / *
- Test / * 
- Deploy / *

### CODE REVIEW

**Review by efalkner — DISMISSED** [review]
LGTM

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+        with:
+          lane: 'staging'
+          subdirectory: 'ios'
+          bundle-install-path: 'vendor/bundle'
+
+      - name: Upload Artifact
+        uses: actions/upload-artifact@v1
+        with:
+          name: COVIDSafePaths-staging-gps
+          path: ios/COVIDSafePaths-staging.ipa
```
Comment (efalkner) [inline .github/workflows/build.yml:223]: should this artifact have gps in the name?

Comment (tstirrat) [reply .github/workflows/build.yml:223]: yep!

Comment (tstirrat) [reply .github/workflows/build.yml:223]: For another PR I guess

**Review by efalkner — APPROVED** [review]
decided to rename artifact in a separate PR ✅

**Pull request merged**